### ISSUE TITLE
Class Imbalance problem

### ISSUE THREAD

**Issue by suryapa1** [ThilinaRajapakse/simpletransformers#280]
TO handle highly imbalanced dataset, like 1:10 ratio of positive classes, what are good parameters to play with. One thing, I tried was to reduce the learning rate and have weight ratio in place. Anything suggestions on rest of arguments to play with imbalance data for binary classification using BERT.

How manage weight decay, gradient accumulation and others from default arguments .

Most of the time, classifying all classes into the majority class, How to handle this scenario?

**Comment by pHequals7** [comment]
commenting here since facing the same problem

@ThilinaRajapakse any help would be much appreciated

**Comment by ThilinaRajapakse** [comment]
Looks like I missed this somehow, sorry.

Class weights are what you need to play with. I can't imagine the other hyperparameters having much effect on class imbalance issues. If weights don't help, you can try to undersample the majority class as well. This might work as BERT doesn't need _too_ many fine-tuning examples.
